src/test/bridges/liquity/TroveBridgeE2E.t.sol

```diff
@@ -0,0 +1,105 @@
+    uint256 private id;
+
+    function setUp() public {
+        // Deploy a new trove bridge
+        vm.prank(OWNER);
+        bridge = new TroveBridge(address(ROLLUP_PROCESSOR), 160);
+
+        // Use the label cheat-code to mark the address with "Trove Bridge" in the traces
+        vm.label(address(bridge), "TroveBridge");
+
+        _baseSetUp();
+
+        // Impersonate the multi-sig to add a new bridge
+        vm.startPrank(MULTI_SIG);
+
+        // List trove bridge with a gasLimit of 500k
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 10000000);
+
+        // List the assets with a gasLimit of 100k
+        ROLLUP_PROCESSOR.setSupportedAsset(tokens["LUSD"].addr, 100000);
+        ROLLUP_PROCESSOR.setSupportedAsset(address(bridge), 100000);
+
+        vm.stopPrank();
+
+        // Fetch the id of the trove bridge
+        id = ROLLUP_PROCESSOR.getSupportedBridgesLength();
+
+        _openTrove();
+    }
+
+    // @dev In order to avoid overflows we set _depositAmount to be uint96 instead of uint256.
+    function testFullFlow(uint96 _collateral) public {
+        uint256 collateral = bound(_collateral, 1e17, 1e21);
+
+        // Use the helper function to fetch Aztec assets
+        AztecTypes.AztecAsset memory ethAsset = getRealAztecAsset(address(0));
+        AztecTypes.AztecAsset memory tbAsset = getRealAztecAsset(address(bridge));
+        AztecTypes.AztecAsset memory lusdAsset = getRealAztecAsset(tokens["LUSD"].addr);
+
+        // 1st BORROW
+        // Mint the collateral amount of ETH to rollupProcessor
+        vm.deal(address(ROLLUP_PROCESSOR), collateral);
+
+        // Compute borrow calldata
+        uint256 bridgeCallData = encodeBridgeCallData(id, ethAsset, emptyAsset, tbAsset, lusdAsset, MAX_FEE);
+
+        uint256 processorLusdBalanceBefore = tokens["LUSD"].erc.balanceOf(address(ROLLUP_PROCESSOR));
+        (uint256 debtBeforeBorrowing, uint256 collBeforeBorrowing, , ) = TROVE_MANAGER.getEntireDebtAndColl(
+            address(bridge)
+        );
+
+        sendDefiRollup(bridgeCallData, collateral);
+
+        (uint256 debtAfterBorrowing, uint256 collAfterBorrowing, , ) = TROVE_MANAGER.getEntireDebtAndColl(
+            address(bridge)
+        );
+        assertEq(
+            collAfterBorrowing - collBeforeBorrowing,
+            collateral,
+            "Collateral increase differs from deposited collateral"
+        );
+        uint256 tbBalanceAfterBorrowing = bridge.balanceOf(address(ROLLUP_PROCESSOR));
+        assertEq(
+            debtAfterBorrowing - debtBeforeBorrowing,
+            tbBalanceAfterBorrowing,
+            "Debt increase differs from processor's TB balance"
+        );
+        assertEq(
+            tokens["LUSD"].erc.balanceOf(address(ROLLUP_PROCESSOR)) - processorLusdBalanceBefore,
+            bridge.computeAmtToBorrow(collateral),
+            "Borrowed amount doesn't equal expected borrow amount"
+        );
+
+        // 2nd REPAY
```

Comment:
Same as above

Reply:
Addressed in 7c41b117bc20dcd19eacc48b4dad738d6cba62e8